### PR TITLE
Split random.dm into categorized files

### PR DESCRIPTION
**Pull request by DreamySkrell**
no user-facing change

### CODE REVIEW

**Comment by DreamySkrell** [comment]
!review

**Comment by DreamySkrell** [comment]
exoplanet_testing randomly failing cause they're quirky like that

File: code/game/objects/random/weapon.dm [hunk 1]
```diff
@@ -0,0 +1,246 @@
+
+/obj/random/energy
+	name = "Random Energy Weapon"
+	desc = "This is a random security weapon."
```
Comment (SleepyGemmy) [inline code/game/objects/random/weapon.dm:4]:
```suggestion
	name = "random energy weapon"
	desc = "This is a random energy weapon."
```

File: code/game/objects/random/weapon.dm [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+	icon_state = "energykill100"
+	problist = list(
+		/obj/item/gun/energy/rifle/laser = 2,
+		/obj/item/gun/energy/gun = 2,
+		/obj/item/gun/energy/stunrevolver = 1
+	)
+
+/obj/random/projectile
+	name = "Random Projectile Weapon"
+	desc = "This is a random security weapon."
```
Comment (SleepyGemmy) [inline code/game/objects/random/weapon.dm:15]:
```suggestion
	name = "random projectile weapon"
	desc = "This is a random projectile weapon."
```

File: code/game/objects/random/weapon.dm [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+	icon_state = "cshotgun"
+	problist = list(
+		/obj/item/gun/projectile/shotgun/pump = 3,
+		/obj/item/gun/projectile/automatic/wt550 = 2,
+		/obj/item/gun/projectile/shotgun/pump/combat = 1
+	)
+
+/obj/random/handgun
+	name = "Random Handgun"
+	desc = "This is a random security sidearm."
```
Comment (SleepyGemmy) [inline code/game/objects/random/weapon.dm:26]:
```suggestion
	name = "random handgun"
	desc = "This is a random handgun."
```

File: code/game/objects/random/weapon.dm [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+	icon = 'icons/obj/guns/secgun.dmi'
+	icon_state = "secgun"
+	problist = list(
+		/obj/item/gun/projectile/sec = 3,
+		/obj/item/gun/projectile/sec/wood = 1
+	)
+
+/obj/random/ammo
+	name = "Random Ammunition"
+	desc = "This is random ammunition."
```
Comment (SleepyGemmy) [inline code/game/objects/random/weapon.dm:36]:
```suggestion
	name = "random ammunition"
	desc = "This is some random ammunition."
```

File: code/game/objects/random/weapon_and_ammo.dm [hunk 5]
```diff
@@ -0,0 +1,185 @@
+
+/obj/random/weapon_and_ammo
+	name = "random weapon and ammo"
+	desc = "Summons a random weapon, with ammo if applicable"
```
Comment (SleepyGemmy) [inline code/game/objects/random/weapon_and_ammo.dm:4]:
```suggestion
	desc = "Summons a random weapon, with ammo if applicable."
```

**Review by SleepyGemmy — DISMISSED** [review]

File: code/game/objects/random/misc.dm [hunk 6]
```diff
@@ -0,0 +1,420 @@
...
+		/obj/item/toy/plushie/beepsky,
+		/obj/item/toy/plushie/ivancarp,
+		/obj/item/toy/plushie/nymph,
+		/obj/item/toy/plushie/mouse,
+		/obj/item/toy/plushie/kitten,
+		/obj/item/toy/plushie/lizard,
+		/obj/item/toy/plushie/farwa,
+		/obj/item/toy/plushie/squid,
+		/obj/item/toy/plushie/bear,
+		/obj/item/toy/plushie/bearfire
```
Comment (SleepyGemmy) [inline code/game/objects/random/misc.dm:86]:
```suggestion
		/obj/item/toy/plushie/bearfire,
		/obj/item/toy/plushie/herring_gull
```
🥺

File: code/game/objects/random/clothing.dm [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+		/obj/item/storage/backpack/duffel/syndie = 1,
+		/obj/item/storage/backpack/messenger = 2,
+		/obj/item/storage/backpack/messenger/pharm = 2,
+		/obj/item/storage/backpack/messenger/med = 2,
+		/obj/item/storage/backpack/messenger/tox = 2,
+		/obj/item/storage/backpack/messenger/com = 1,
+		/obj/item/storage/backpack/messenger/engi = 2,
+		/obj/item/storage/backpack/messenger/hyd = 2,
+		/obj/item/storage/backpack/messenger/sec = 2,
+		/obj/item/storage/backpack/messenger/syndie = 1,
```
[the remark this answers is not in the document]
Comment (DreamySkrell) [reply code/game/objects/random/clothing.dm:60]: I disagree with this coding style suggestion.
No comma at end is ass, and only causes annoying errors in the future, where you copy the entry from the end to another list, and need to manually add the comma then.
Comma at end is standard in a lot of newer languages like flutter/dart, or rust.

File: code/game/objects/random/clothing.dm [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+/obj/random/watches
+	name = "random watches"
+	desc = "Random watches, probably able to tell the time."
+	icon = 'icons/obj/clothing/gloves.dmi'
+	icon_state = "watch"
+	problist = list(
+		/obj/item/clothing/wrists/watch = 1,
+		/obj/item/clothing/wrists/watch/silver = 0.7,
+		/obj/item/clothing/wrists/watch/gold = 0.5,
+		/obj/item/clothing/wrists/watch/spy = 0.3,
```
[the remark this answers is not in the document]
Comment (DreamySkrell) [reply code/game/objects/random/clothing.dm:171]: I disagree with this coding style suggestion.
No comma at end is ass, and only causes annoying errors in the future, where you copy the entry from the end to another list, and need to manually add the comma then.
Comma at end is standard in a lot of newer languages like flutter/dart, or rust.

**Comment by SleepyGemmy** [comment]
i deleted some code review comments from me and that also deletes any replies, as context to the above.

**Review by Arrow768 — APPROVED** [review]

**Comment by DreamySkrell** [comment]
merge conflict fixed

**Review by NonQueueingMatt — APPROVED** [review]

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**